src/_P036_FrameOLED.ino

```diff
@@ -980,37 +1035,73 @@ byte display_scroll(int lscrollspeed, int lTaskTimer)
   else iPageScrollTime = (P36_MaxDisplayWidth /(P36_PageScrollPix * lscrollspeed)) * P36_PageScrollTick;
   float fScrollTime = (float)(lTaskTimer*1000 - iPageScrollTime - 2*P36_WaitScrollLines*100)/100.0;
 
-// log += F(" PageScrollTime: ");
-// log += iPageScrollTime;
+  // log += F(" PageScrollTime: ");
+  // log += iPageScrollTime;
 
-  for (byte j = 0; j < ScrollingPages.linesPerFrame; j++)
-  {
-    // settings for following line scrolling
-    ScrollingLines.Line[j].LastWidth = ScrollingLines.Line[j].Width;  // save last line width
-    word PixLength = display->getStringWidth(ScrollingPages.newString[j]);
-    if ((PixLength > SizeSettings[OLEDIndex].Width) && (fScrollTime > 0.0))
-    {
-      // width of the line > display width -> scroll line
-      ScrollingLines.Line[j].Content = ScrollingPages.newString[j];
-      ScrollingLines.Line[j].Width = PixLength;
-      ScrollingLines.Line[j].CurrentLeft = SizeSettings[OLEDIndex].PixLeft;
-      ScrollingLines.Line[j].fPixSum = (float) SizeSettings[OLEDIndex].PixLeft;
-      ScrollingLines.Line[j].dPix = ((float)(PixLength-SizeSettings[OLEDIndex].Width))/fScrollTime; // pix change per scrolling line tick
-
-// log += F(" line: ");
-// log += j+1;
-// log += F(" width: ");
-// log += ScrollingLines.Line[j].Width;
-// log += F(" dPix: ");
-// log += ScrollingLines.Line[j].dPix;
-    }
-    else {
-      ScrollingLines.Line[j].Width = 0; // do not scroll line
+    word MaxPixWidthForPageScrolling = P36_MaxDisplayWidth;
+    if (bScrollLines) {
+      // Reduced scrolling width because line is displayed left or right aligned
+      MaxPixWidthForPageScrolling -= SizeSettings[OLEDIndex].PixLeft;
     }
+
+    for (byte j = 0; j < ScrollingPages.linesPerFrame; j++)
+    {
+      // default no line scrolling and strings are centered
+      ScrollingLines.Line[j].LastWidth = 0;
+      ScrollingLines.Line[j].Width = 0;
+
+      // get last and new line width
+      word LastPixLength = display->getStringWidth(ScrollingPages.oldString[j]);
+      word NewPixLength = display->getStringWidth(ScrollingPages.newString[j]);
```

Comment:
Please use int32_t or uint32_t kind of types.
I always find it hard to remember what bit depth a word or a short is. (and whether it is signed or unsigned)

Reply:
Done!